### PR TITLE
Bundle validator

### PR DESCRIPTION
**Pull request by phildow**
.tfbundle validation that supports downloading and importing models directly from within Net Runner. The bulk of the code is JSON validation and error reporting, trying to provide helpful feedback about errors in the model.json file. Test coverage for the class is extensive.

### CODE REVIEW

File: Example/Tests/TensorIOModelBundleValidatorTests.mm [hunk 1]
```diff
@@ -0,0 +1,2174 @@
...
+    
+    BOOL valid = [self.modelValidator validateBundleProperties:properties error:&error];
+    
+    XCTAssertTrue(valid);
+    XCTAssertNil(error);
+}
+
+// MARK: - Model Properties Validation
+
+- (void)testModelPropertiesValidates {
```
Comment (nkashy1) [inline Example/Tests/TensorIOModelBundleValidatorTests.mm]: Would split this up into different tests.

File: Example/Tests/TensorIOModelBundleValidatorTests.mm [hunk 2]
```diff
@@ -0,0 +1,2174 @@
...
+    
+    valid = [self.modelValidator validateModelProperties:properties error:&error];
+    
+    XCTAssertTrue(valid);
+    XCTAssertNil(error);
+}
+
+// MARK: - Assets Validation
+
+- (void)testAnIncorrectlyNamedModelFileDoesNotValidate {
```
Comment (nkashy1) [inline Example/Tests/TensorIOModelBundleValidatorTests.mm]: Can we add model file validation to check that it is a valid TFLite file? This is not a blocker on this PR, and I'm not even sure if I would want it as a user (if, for example, it required us to run dummy data through a model). Just something to consider.

I don't know if there's an easy way (like `saved_model_cli` binary which comes with TensorFlow) to test TFLite models.

File: Example/Tests/TensorIOModelBundleValidatorTests.mm (hunk 1 above)
Comment (nkashy1) [reply Example/Tests/TensorIOModelBundleValidatorTests.mm]: Again, not blocker and it's largely a matter of preference -- just if the tests ever fail, it's more informative if the tests are smaller.

File: Example/Tests/TensorIOModelBundleValidatorTests.mm (hunk 2 above)
Comment (phildow) [reply Example/Tests/TensorIOModelBundleValidatorTests.mm]: Let's add this to 0.3.x. I'll make an issue for it. If nothing else we can try to load the model and if that fails report that to the user.

**Comment by phildow** [comment]
@nkashy1 We could compile this into a command line utility, yes. I was also thinking we might write a little python script to generate the model.json file. I think that would be useful for data scientists, with the added risk that we now maintain a couple of different code bases, albeit small ones, one python and the other objc.

**Pull request merged**